wdl-analysis/src/handlers/completions.rs

```diff
@@ -0,0 +1,720 @@
+                };
+
+                items.push(CompletionItem {
+                    label: name.to_string(),
+                    kind,
+                    detail,
+                    ..Default::default()
+                });
+            }
+        }
+        current_scope = s.parent();
+    }
+}
+
+/// Adds completions for all WDL standard library functions.
+fn add_stdlib_completions(items: &mut Vec<CompletionItem>) {
+    for (name, func) in STDLIB.functions() {
+        match func {
+            Function::Monomorphic(m) => {
+                let sig = m.signature();
+                let params = TypeParameters::new(sig.type_parameters());
+                let detail = Some(format!("{name}{}", sig.display(&params)));
+                let docs = sig.definition().and_then(|d| make_docs(d.to_string()));
+                items.push(CompletionItem {
+                    label: name.to_string(),
+                    kind: Some(CompletionItemKind::FUNCTION),
+                    detail,
+                    documentation: docs,
+                    ..Default::default()
+                })
+            }
+            Function::Polymorphic(p) => {
+                for sig in p.signatures() {
+                    let params = TypeParameters::new(sig.type_parameters());
+                    let detail = Some(format!("{name}{}", sig.display(&params)));
+                    let docs = sig.definition().and_then(|d| make_docs(d.to_string()));
+                    items.push(CompletionItem {
+                        label: name.to_string(),
+                        kind: Some(CompletionItemKind::FUNCTION),
+                        detail,
+                        documentation: docs,
+                        ..Default::default()
+                    });
+                }
+            }
+        };
+    }
+}
+
+/// Adds completions for user-defined structs in the document.
+fn add_struct_completions(document: &Document, items: &mut Vec<CompletionItem>) {
+    let root = document.root();
+    for (name, s) in document.structs() {
+        items.push(CompletionItem {
+            label: name.to_string(),
+            kind: Some(CompletionItemKind::STRUCT),
+            detail: Some(format!("struct {name}")),
+            documentation: provide_struct_documentation(s, &root).and_then(make_docs),
+            ..Default::default()
+        })
+    }
+}
+
+/// Adds completions for imported namespaces (aliases).
+fn add_namespace_completions(document: &Document, items: &mut Vec<CompletionItem>) {
+    for (name, _) in document.namespaces() {
+        items.push(CompletionItem {
+            label: name.to_string(),
+            kind: Some(CompletionItemKind::MODULE),
+            detail: Some(format!("import alias {name}")),
+            ..Default::default()
+        });
+    }
+}
+
+/// Adds completions for the members of the implicit `task` variable.
+fn add_task_variable_completions(items: &mut Vec<CompletionItem>) {
+    const TASK_FIELDS: &[&str] = &[
```

Comment:
For every field that is a `const` like this, I'd prefer if we defined the arrays in `wdl-grammar` right next to the keywords. The reason for this is I want all of this kind of information to be stored in one part of the codebase. It's much more likely that, if, for example, a new keyword is added in the future, the engineer will update the list if it lives there than if it is buried here in the LSP code. We can still import it and use it here obviously.